Fixing Inconsistencies in Chat events

grangasamy-msft opened the pull request:
# Data Plane API Specification Update Pull Request

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.


<!-- 🚨🚨🚨 Replace this line with a summary and reason for these changes to your data plane API 🚨🚨🚨 -->

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## API Info: The Basics

Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Schedule a data plane REST API spec review](https://aka.ms/azsdk/onboarding/restapischedule)

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories) 
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

grangasamy-msft committed:
Fixing Inconsistencies in Chat events

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Microsoft.ApiCenter](https://apiview.dev/Assemblies/Review/3c9fafea50ce44d9a36a97c2e9751516?revisionId=8c1dd6f15f204fe6a5c1f8d2943ef835)

grangasamy-msft committed:
Added ChatAzureBotCommand event.

grangasamy-msft committed:
Added ChatTypingIndication event to SDK.

grangasamy-msft committed:
Fixing ChatTyping name from ChatTypingIndicator to ChatTypingIndicatorSent

l0lawrence commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -971,10 +996,10 @@ model AcsChatMessageEventInThreadBaseProperties
 
   /** The version of the message */
-  version?: int64;
+  version: int64;
```
this needs @madeRequired b/c existing property that is a value type

grangasamy-msft committed:
Bug fixes

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -128,7 +146,7 @@ model AcsChatThreadPropertiesUpdatedPerUserEventData
 
   /** The thread metadata */
-  metadata: Record<string>;
+  metadata?: Record<string>;
```
Why is this changing to optional?

grangasamy-msft replied:
Thats kind of what I got from the chat team

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -934,7 +952,8 @@ model AcsChatThreadEventInThreadBaseProperties
 model AcsChatMessageEventBaseProperties extends AcsChatEventBaseProperties {
   /** The chat message id */
-  messageId?: string;
+  @madeRequired(ServiceApiVersions.v2024_01_01)
```
This doesn't need to be made required as it is a string. It should also be added to required list in the swagger.

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -947,18 +966,21 @@ model AcsChatMessageEventBaseProperties extends AcsChatEventBaseProperties {
   extends AcsChatEventInThreadBaseProperties {
   /** The chat message id */
-  messageId?: string;
+  @madeRequired(ServiceApiVersions.v2024_01_01)
```
Doesn't need maderequired.

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -983,7 +1005,8 @@ model AcsChatEventInThreadBaseProperties {
 
   /** The chat thread id */
-  threadId?: string;
+  @madeRequired(ServiceApiVersions.v2024_01_01)
```
Doesn't need maderequired.

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -1080,24 +1103,24 @@ union AcsUserEngagement {
 
   /** The Microsoft Teams user. */
-  microsoftTeamsUser: MicrosoftTeamsUserIdentifierModel;
+  microsoftTeamsUser?: MicrosoftTeamsUserIdentifierModel;
```
Why is this changing?

grangasamy-msft replied:
Thats kind of what I got from Chat team.

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -1080,24 +1103,24 @@ union AcsUserEngagement {
 
   /** The phone number. */
-  phoneNumber: PhoneNumberIdentifierModel;
+  phoneNumber?: PhoneNumberIdentifierModel;
```
Why is this changing?

grangasamy-msft replied:
Thats kind of what I got from chat team

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -1080,24 +1103,24 @@ union AcsUserEngagement {
 
   /** Raw Id of the identifier. Optional in requests, required in responses. */
-  rawId?: string;
+  @madeRequired(ServiceApiVersions.v2024_01_01)
```
Doesn't need maderequired.

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -43,6 +43,24 @@ model AcsChatMessageReceivedEventData
 
+/** Schema of the Data property of an EventGridEvent for a Microsoft.Communication.ChatAzureBotCommand event. */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "Maintain compatibility with existing models."
+model AcsChatAzureBotCommandEventData
```
Usually events are named as verbs, e.g. AcsChatThreadCreatedEventData, AcsChatMessageDeletedEventData, as they are describing an event that occurred in the system. What is the event here?

grangasamy-msft replied:
AcsChatAzureBotCommandReceivedInThread

grangasamy-msft committed:
Updated event type to new names
ChatTypingIndicatorReceivedInThread and ChatAzureBotCommanReceivedInThread, and made changes accordingly.
